Userland/Applications/Dictionary/MainWidget.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2021, Robin Allen <r@foon.uk>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "MainWidget.h"
+
+#include "DictionaryModel.h"
+
+#include <Applications/Dictionary/DictionaryWindowGML.h>
+#include <LibGUI/ListView.h>
+#include <LibGUI/TextBox.h>
+#include <LibGUI/TextEditor.h>
+
+namespace Dictionary {
+
+MainWidget::MainWidget(char const* initial_query)
```

Comment:
Suggest taking a StringView here so that you don't have to strnlen the query to set it as the text for the search box and the model's query.

Reply:
Done.